src/core/vote-core.js

```diff
@@ -0,0 +1,82 @@
+import _ from 'lodash';
+const {knex} = require('../util/database').connect();
+import {hotScore} from '../sort';
+import {actionToFeedObject} from './feed-core';
+
+function createOrUpdateVote(opts) {
+
+  const updateVoteSql = `
+    WITH upsert AS
+      (UPDATE votes SET value = ? WHERE user_id = ? AND feed_item_id = ? RETURNING *)
+      INSERT INTO votes (value, user_id, feed_item_id)
+      SELECT ?, ?, ? WHERE NOT EXISTS (SELECT * FROM upsert)
+  `;
+
+  const selectVotesSql = `
+    SELECT
+      feed_items.id as id,
+      votes(feed_items) as votes,
+      EXTRACT (EPOCH FROM feed_items.created_at) as age
+    FROM feed_items
+    WHERE feed_items.id = ?
+    GROUP BY feed_items.id
+  `;
+
+  const updateFeedItemSql = `
+    UPDATE
+      feed_items
+    SET
+      hot_score = ?
+    FROM feed_items as feed_item
+    LEFT JOIN users ON users.id = feed_item.user_id
+    LEFT JOIN teams ON teams.id = users.team_id
+    WHERE
+      feed_items.id = ? AND feed_items.id = feed_item.id
+    RETURNING
+      feed_items.id as id,
+      feed_items.location as location,
+      feed_items.created_at as created_at,
+      feed_items.image_path as image_path,
+      feed_items.text as text,
+      feed_items.type as action_type_code,
+      COALESCE(users.name, 'SYSTEM') as user_name,
+      users.uuid as user_uuid,
+      teams.name as team_name,
+      votes(feed_items) as votes,
+      feed_items.hot_score as hot_score,
+      feed_items.is_sticky
```

Comment:
Do we really need to select all this stuff in UPDATE? If we change the API to not return the updated feed item but instead just some OK status, we could get rid of this select and LEFT JOINs in the UPDATE statement resulting in simpler and faster code.

From the client perspective it should do an optimistic update and update score based on the current user's vote. Then after the feed is refreshed the new score counts can be updated. (At least personally I would find it confusing if I vote an item up and it suddenly the score goes up e.g. 5 points, or e.g. down 1 point)